### PR TITLE
Banner donnation december

### LINKED ISSUE #1386: Feature: December Donation Banner
## Background Information
In December, we will run an email campaign for donation that will happen yearly

## What
- A banner that always appear at the top of the AletheiaFact.org website. 
- The user can close it by clicking in the "X" and we save a cookie to respect that choice
- Use a feature flag to enable it and another one for A/B testing
- The A/B testing will just change the text content
- Track the click of each banner considering if the user clicked on A or B

### PR DESCRIPTION
**Pull request by LuizFNJ**
# Description
*Donation Banner December Campaign*
*home:*
Screenshot from 2024-11-06 15-34-54

*on any other page there is a header:*
Screenshot from 2024-11-06 15-35-15

Fixes #1386 

# Questions
- *The hide button is necessary?*
- *When the user is not logged in on the homepage, there will be 3 donation buttons. I would suggest hiding the donation button in the header when the banner is being displayed.*

# Note
*I haven't done the responsiveness for smaller screens yet. I'm facing difficulties in the development environment because the layout is breaking at 768px, regardless of the changes I make.*

### CODE REVIEW

File: src/components/MainApp.tsx [hunk 1]
```diff
@@ -43,6 +44,7 @@ const MainApp = ({ children }) => {
             <Sidebar />
             <Layout style={{ background: colors.white }}>
                 <Header />
+                <DonationBanner />
```
Comment (thesocialdev) [inline src/components/MainApp.tsx:47]: chore: before this is ready it needs to have a feature flag

**Review by thesocialdev — CHANGES_REQUESTED** [review]

File: src/components/MainApp.tsx (hunk 1 above)
Comment (LuizFNJ) [reply src/components/MainApp.tsx]: This is the first time I've done this, if you can confirm if this is correct - @pepermao @caneppelevitor @thesocialdev 

commit:
13d1b29427d4e11e9cf502b8de2eea065fee6604

File: src/components/Home/DonationBanner/DonationBannerContent.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            >
+                <h1>
+                    {t("donationBanner:title", {
+                        date: new Date().getFullYear(),
+                    })}
+                </h1>
+                <p>
+                    {t("donationBanner:paragraph")}
+                </p>
+                <BannerButton type={ButtonType.blue} />
```
Comment (thesocialdev) [inline src/components/Home/DonationBanner/DonationBannerContent.tsx:23]: todo: we also need to register a click on this button for analytics purposes.

File: src/components/Home/DonationBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React, { useState } from "react";
+import DonationBannerContent from "./DonationBanner/DonationBannerContent";
+import DonationBannerStyle from "./DonationBanner.style";
+import { Col } from "antd";
+import { useTranslation } from "next-i18next";
+
+const DonationBanner = () => {
+  const { t } = useTranslation();
+  const [isBannerVisible, setIsBannerVisible] = useState(true);
+  const enableDonationBanner = process.env.NEXT_PUBLIC_ENABLE_BANNER_DONATION === "true";
```
Comment (thesocialdev) [inline src/components/Home/DonationBanner.tsx:10]: suggestion: this is looking great! now we need to create the deployment changes to include that variable in the containers

**Review by thesocialdev — CHANGES_REQUESTED** [review]

File: .github/workflows/aws.yml [hunk 4]
```diff
@@ -34,6 +34,7 @@ jobs:
       ORY_SDK_URL: ${{ secrets.ORY_SDK_URL }}
       ORY_ACCESS_TOKEN: ${{ secrets.ORY_ACCESS_TOKEN }}
       ORYCLOUD: "enabled"
+      ENABLE_BANNER_DONATION: ${{ env.NEXT_PUBLIC_ENABLE_BANNER_DONATION }}
```
Comment (thesocialdev) [inline .github/workflows/aws.yml:37]: question: should it be `vars` instead of `env`?

File: Dockerfile [hunk 5]
```diff
@@ -6,6 +6,7 @@ ARG NEXT_PUBLIC_ENVIRONMENT
 ARG NEXT_PUBLIC_ORY_SDK_URL
 ARG NEXT_PUBLIC_ORYCLOUD
 ARG ENVIRONMENT
+ENV NEXT_PUBLIC_ENABLE_BANNER_DONATION=false
```
Comment (thesocialdev) [inline Dockerfile:9]: issue: should be ARG and be passed in the build

File: .github/workflows/aws.yml (hunk 4 above)
Comment (LuizFNJ) [reply .github/workflows/aws.yml]: it has to be vars, because that's how we're going to use it in github actions. I got confused

**Review by thesocialdev — APPROVED** [review]

**Pull request merged**